ethstaker_deposit/cli/generate_keys.py

```diff
@@ -120,12 +129,27 @@ def generate_keys_arguments_decorator(function: Callable[..., Any]) -> Callable[
 @click.pass_context
 def generate_keys(ctx: click.Context, validator_start_index: int,
                   num_validators: int, folder: str, chain: str, keystore_password: str,
-                  withdrawal_address: HexAddress, pbkdf2: bool, **kwargs: Any) -> None:
+                  withdrawal_address: HexAddress, pbkdf2: bool,
+                  devnet_chain_setting: Optional[str], **kwargs: Any) -> None:
     mnemonic = ctx.obj['mnemonic']
     mnemonic_password = ctx.obj['mnemonic_password']
     amounts = [MIN_ACTIVATION_AMOUNT] * num_validators
     folder = os.path.join(folder, DEFAULT_VALIDATOR_KEYS_FOLDER_NAME)
-    chain_setting = get_chain_setting(chain)
+
+    # Get chain setting
+    if devnet_chain_setting is not None:
+        click.echo('\n%s\n' % load_text(['arg_devnet_chain_setting', 'warning'],
```

Comment:
When generating a new mnemonic, this message is hidden due to the newlines created unless you have a very long terminal

Reply:
Given the code duplication here, could we move this logic to `validation.py` `validate_devnet_chain_setting` and instead of returning the `trimmed_value`, we can use that to build out the `devnet_chain_setting`.

Then instead of returning `Optional[str]`, it would return `Optional[BaseChainSetting]` and we would update the logic here to something like

`chain_setting = devnet_chain_setting if devnet_chain_setting is not None else get_chain_setting(chain)`

This would also make the warning message visible during prompts.

If we do move the logic, would also like to see a unit test or two to test the basic flow